pkg/splunk/enterprise/standalone.go

```diff
@@ -67,12 +69,22 @@ func ApplyStandalone(client splcommon.ControllerClient, cr *enterprisev1.Standal
 		cr.Status.SmartStore = cr.Spec.SmartStore
 	}
 
-	// ToDo sgontla: Handle
-	// 1. Spec update
-	// 2. S3 credentails change
-	// 3. Probe times expired
-	if !reflect.DeepEqual(cr.Status.AppContext.AppFrameworkConfig, cr.Spec.AppFrameworkConfig) {
-		// TBD: Probe the remote store, and handle any changes on remote store
+	if cr.Spec.CommonSplunkSpec.Mock != true && !reflect.DeepEqual(cr.Status.AppContext.AppFrameworkConfig, cr.Spec.AppFrameworkConfig) {
+		var sourceToAppsList map[string]splclient.S3Response
+
+		for _, vol := range cr.Spec.AppFrameworkConfig.VolList {
```

Comment:
For clustermaster we have the following snippet of code:

```
for _, vol := range cr.Spec.AppFrameworkConfig.VolList {			
    if _, ok := splclient.S3Clients[vol.Provider]; !ok  {
				splclient.RegisterS3Client(vol.Provider)			
     }
}
```

Are we skipping the check on the entries in the map `splclient.S3Clients`

Reply:
Udpated on other CRs as well!

Reply:
@gaurav-splunk , I still see a difference with clustermaster.go. Am I missing something here?

Reply:
Yeah you are missing my update commit which I haven't pushed yet :)

Reply:
I'll resolve it now and approve it then :)

Reply:
Done!